tests/components/homee/test_config_flow.py

```diff
@@ -130,3 +130,108 @@ async def test_flow_already_configured(
     )
     assert result["type"] is FlowResultType.ABORT
     assert result["reason"] == "already_configured"
+
+
+@pytest.mark.usefixtures("mock_setup_entry")
+async def test_reconfigure_success(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_homee: AsyncMock,
+) -> None:
+    """Test the reconfigure flow."""
+    mock_config_entry.add_to_hass(hass)
+    mock_config_entry.runtime_data = mock_homee
+    result = await mock_config_entry.start_reconfigure_flow(hass)
+
+    assert result["step_id"] == "reconfigure"
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {}
+    assert result["handler"] == DOMAIN
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_HOST: NEW_HOMEE_IP,
+        },
+    )
+
+    assert result2["type"] is FlowResultType.ABORT
+    assert result2["reason"] == "reconfigure_successful"
+
+    # Confirm that the config entry has been updated
+    assert mock_config_entry.data[CONF_HOST] == NEW_HOMEE_IP
+
+
+@pytest.mark.parametrize(
+    ("side_eff", "error"),
+    [
+        (
+            HomeeConnectionFailedException("connection timed out"),
+            {"base": "cannot_connect"},
+        ),
+        (
+            HomeeAuthFailedException("wrong username or password"),
+            {"base": "invalid_auth"},
+        ),
+        (
+            Exception,
+            {"base": "unknown"},
+        ),
+    ],
+)
+async def test_reconfigure_errors(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_homee: AsyncMock,
+    side_eff: Exception,
+    error: dict[str, str],
+) -> None:
+    """Test reconfigure flow errors."""
+    mock_config_entry.add_to_hass(hass)
+    mock_config_entry.runtime_data = mock_homee
+    result = await mock_config_entry.start_reconfigure_flow(hass)
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "reconfigure"
+
+    mock_homee.get_access_token.side_effect = side_eff
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_HOST: NEW_HOMEE_IP,
+        },
+    )
+
+    assert result2["type"] == FlowResultType.FORM
```

Comment:
```suggestion
    assert result2["type"] is FlowResultType.FORM
```
for the others as well